lib/Parser.js

```diff
@@ -169,22 +170,117 @@ Parser.prototype = {
                     results = this._transforms.produce(options.transform, results);
                 }
             })
-            .then(this._env.tearDown, e => {
-                return this._env
+            .then(
+                () => this._env.tearDown(),
+                e => this._env
                     .snapshot('error')
                     .always(this._env.tearDown, this._env)
                     .then(() => {
                         throw e
-                    });
-            }, this._env)
-            .then(function() {
-                debug('Parsed %s rows', Array.isArray(results) ? results.length : 1);
-                return results;
-            }, function(e) {
-                debug('Saving partial results because of error %o', e);
-                this._partialResults = results;
-                throw e;
-            }, this);
+                    })
+            )
+            .then(
+                () => {
+                    debug('Parsed %s rows', Array.isArray(results) ? results.length : 1);
+                    return results;
+                },
+                e => {
+                    debug('Saving partial results because of error %o', e);
+                    this._partialResults = results;
+                    throw e;
+                });
+    },
+
+    moveYourFeet(stages) {
+        debug('Hit the road!');
+        const milestones = stages.milestones;
+        const edgeCases = stages.edgeCases;
+
+        return this._env
+            .prepare()
+            .then(() => milestones.reduce((promise, milestone) => {
+                return promise.then(() => this.passMilestone(milestone, edgeCases));
+            }, Promise.resolve()))
+            .then(
+                () => this._env.tearDown(),
+                e => this._env
+                    .snapshot('error')
```

Comment:
this is not a good idea to do it all time, it's quite lazy operation.
Let's use it, if we have special option in phantomenv, we already have one